src/common/config.cc

```diff
@@ -515,61 +566,70 @@ int md_config_t::parse_option(std::vector<const char*>& args,
     return ret;
   }
 
+  const char *option_name = nullptr;
+  std::string error_message;
   o = 0;
   for (auto& opt_ref: *config_options) {
     ostringstream err;
     config_option const *opt = &opt_ref;
     std::string as_option("--");
     as_option += opt->name;
+    option_name = opt->name;
     if (opt->type == OPT_BOOL) {
       int res;
       if (ceph_argparse_binary_flag(args, i, &res, oss, as_option.c_str(),
 				    (char*)NULL)) {
 	if (res == 0)
-	  set_val_impl("false", opt);
+	  ret = set_val_impl("false", opt, &error_message);
 	else if (res == 1)
-	  set_val_impl("true", opt);
+	  ret = set_val_impl("true", opt, &error_message);
 	else
 	  ret = res;
 	break;
       } else {
 	std::string no("--no-");
 	no += opt->name;
 	if (ceph_argparse_flag(args, i, no.c_str(), (char*)NULL)) {
-	  set_val_impl("false", opt);
+	  ret = set_val_impl("false", opt, &error_message);
 	  break;
 	}
       }
-    }
-    else if (ceph_argparse_witharg(args, i, &val, err,
-				   as_option.c_str(), (char*)NULL)) {
+    } else if (ceph_argparse_witharg(args, i, &val, err,
+                                     as_option.c_str(), (char*)NULL)) {
       if (!err.str().empty()) {
-	*oss << err.str();
+        error_message = err.str();
 	ret = -EINVAL;
 	break;
       }
       if (oss && ((!opt->is_safe()) &&
 		  (observers.find(opt->name) == observers.end()))) {
 	*oss << "You cannot change " << opt->name << " using injectargs.\n";
-	ret = -ENOSYS;
-	break;
-      }
-      int res = set_val_impl(val.c_str(), opt);
-      if (res) {
-	if (oss) {
-	  *oss << "Parse error setting " << opt->name << " to '"
-	       << val << "' using injectargs.\n";
-	  ret = res;
-	  break;
-	} else {
-	  cerr << "parse error setting '" << opt->name << "' to '"
-	       << val << "'\n" << std::endl;
-	}
+        return -ENOSYS;
       }
+      ret = set_val_impl(val, opt, &error_message);
       break;
     }
     ++o;
   }
+
+  if (ret != 0) {
```

Comment:
I see it checks error_message in parse_config_files_impl, so the warning is output when the config file is invalid, but if I specify it via args (`rbd --rbd_default_features=INVALID create ...`) there is no any warning.